python/ray/train/__init__.py

```diff
@@ -25,7 +24,6 @@
     "report",
     "save_checkpoint",
     "TrainingIterator",
-    "TrainingCallback",
```

Comment:
Can we do something like this until we do a full removal? 

```
class TrainingCallback:
    def __init__(self, *args, **kwargs):
        raise DeprecationWarning(...)
```

Reply:
This is a bit tricky because TrainingCallback is an abstract class, meaning it's init will be often overriden, preventing the error from being raised. I'll think of a way to make this work.

Reply:
Added back in using `__new__`.